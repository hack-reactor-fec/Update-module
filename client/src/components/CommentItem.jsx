import React from 'react';
import moment from 'moment';

const CommentItem = (props) => (
    <div className="update-post-comment-item">
        <img className="comment-avatar" src={props.comment.avatar} alt="Smiley face" height="20" width="20" />
        <span className="comment-username">{props.comment.userName}</span>
        <span className="comment-date">{moment(props.comment.date).fromNow()}</span>
        <p className="comment-body">{props.comment.body}</p>
    </div>
)



export default CommentItem;