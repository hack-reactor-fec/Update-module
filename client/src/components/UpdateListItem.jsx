import React from 'react';

const UpdateListItem = (props) => (
    <div id="update-list-item" onClick={() => props.handleClick(props.update.title)}>
        <h5 id="update-date">{props.update.date}</h5>
        <h3 id="update-title">{props.update.title}</h3>
        <p id="update-body">{props.update.body} ...Read More</p>
        <div id="likes-comments">
            <span id="update-comments">{(()=>{
                if (props.update.comments.length > 1) {return props.update.comments.length + ' Comments'}
                else if (props.update.comments.length === 1) {return props.update.comments.length + ' Comment'}
                else {return}})()}
            </span>
            <span id="update-likes"> {(() => {
                if (props.update.likes > 1) {return props.update.likes + ' Likes'}
                else if (props.update.likes === 1) {return props.update.likes + ' Like'}
                else {return}})()}
            </span>
        </div>
    </div>
)

export default UpdateListItem;