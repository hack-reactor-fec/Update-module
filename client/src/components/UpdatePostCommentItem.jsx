import React from 'react';
//use moments

const UpdatePostCommentItem = (props) => (
    <div id="update-post-comment-item">
        <img id="comment-avatar" src={props.comment.avatar} alt="Smiley face" height="20" width="20"/>
        <span id="comment-username">{props.comment.userName}</span>
        <span id="comment-date">{props.comment.date}</span>
        <p id="comment-body">{props.comment.body}</p>
    </div>
)



export default UpdatePostCommentItem;