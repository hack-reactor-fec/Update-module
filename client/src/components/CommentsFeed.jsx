import React from 'react';
import CommentItem from './CommentItem.jsx'

const CommentsFeed = (props) => (
    <div>
        <h4 id="update-post-comments">Comments</h4>
        <h6 id="update-post-comments-login">Only backers can post comments. <span id="login">Log In</span></h6>
        <h6 id="update-post-comments-login">Use this space to cheer the creator along, and talk to your fellow backers.<br/>><br/>>Have a question?<span id="login">Check out the FAQ</span></h6>
        {props.comments.map((comment, i) => <CommentItem comment={comment} key={"c" + i}/>)}
    </div>
)

export default CommentsFeed;