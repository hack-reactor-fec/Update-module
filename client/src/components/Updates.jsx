import React from 'react';
import UpdateListItem from './UpdateListItem.jsx';
import moment from 'moment';

const Updates = (props) => (
    <div>
        {props.updates.map((update, i) => <UpdateListItem update={update} key={"u"+i} handleClick={props.handleClick} />)}
        <div id="project-launched-box">
            <h5 id="created-at">{moment(props.createdAt).format('MMMM D, YYYY')}</h5>
            <h3 id="project-launched">Project Launched</h3>
        </div>
    </div>
)

export default Updates;