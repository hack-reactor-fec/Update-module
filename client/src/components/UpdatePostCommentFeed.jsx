import React from 'react';
import UpdatePostCommentItem from './UpdatePostCommentItem.jsx'

const UpdatePostCommentFeed = (props) => (
    <div>
        <h4 id="update-post-comments">Comments</h4>
        <h6 id="update-post-comments-login">Only backers can post comments. <span id="login">Log In</span></h6>
        {props.comments.map((comment, i)=> <UpdatePostCommentItem comment={comment} key={"uc"+i}/>)}
    </div>
)

export default UpdatePostCommentFeed;