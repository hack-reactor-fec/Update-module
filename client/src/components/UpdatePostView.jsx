import React from 'react';
import UpdatePostCommentFeed from './UpdatePostCommentFeed.jsx'

const UpdatePostView = (props) => (
     <div>
        <h5 id="update-date">{props.update.date}</h5>
        <h2 id="update-title">{props.update.title}</h2>
        <div id="comments-like-bar">
            <span id="update-post-comments">{(() => {
                if (props.update.comments.length > 1) { return props.update.comments.length + ' Comments ' }
                else if (props.update.comments.length === 1) { return props.update.comments.length + ' Comment ' }
                else { return }})()}
            </span>
            <button id="like-button" type="button">♥ Like</button>
            <span id="update-post-likes"> {(() => {
                if (props.update.likes > 1) { return props.update.likes + ' Likes' }
                else if (props.update.likes === 1) { return props.update.likes + ' Like' }
                else { return }})()}
            </span>
        </div>
        <p id="update-body">{props.update.body}</p>
        <div id="previous-next-update-bar">
            <button id="previous">&lt; Previous Update</button>
            <button id="previous">Next Update &gt;</button>
        </div>
        <button id="like-button-2" type="button">♥ Like</button>
        <UpdatePostCommentFeed comments={props.update.comments}/>
    </div>
)

export default UpdatePostView;