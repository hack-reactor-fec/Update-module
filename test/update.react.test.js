import React from 'react';
import App from '../client/src/components/App.jsx';
// import renderer from 'react-test-renderer';
import Updates from '../client/src/components/Updates.jsx';
import UpdateListItem from '../client/src/components/UpdateListItem.jsx';
import UpdatePostCommentFeed from '../client/src/components/UpdatePostCommentFeed.jsx';
import UpdatePostCommentItem from '../client/src/components/UpdatePostCommentItem.jsx';
import UpdatePostView from '../client/src/components/UpdatePostView.jsx';
import CommentsFeed from '../client/src/components/CommentsFeed.jsx';
import CommentItem from '../client/src/components/CommentItem.jsx';

const testData = {
    createdAt: "2018-02-13 20:03:17.035Z",
    updateView: {
        previous: null,
        current: null,
        next: null
    },
    updates: [{
        title: "2nd update",
        body: "OMG THANKS FOR BACKING OMG",
        date: "4/4/2018",
        likes: 420,
        comments: [{
            userId: 2,
            avatar: "https://uproxx.files.wordpress.com/2017/02/kevin-durant-mad-yell.jpg?quality=95",
            userName: "KD",
            date: "3/30/2018",
            body: "u is welcome",
        }]
    }, {
        title: "first update",
        body: "OMG THANKS FOR BACKING OMG!!!!!!!!!!!!!!!!! like omg omgomgomgomg alsjdf asdf jalksd flajsd flkasj dflkaj sldjf alksdjf laksjd flkas fla sdfl alsdf alsjdf laksjdf laksjdf laksjd flkasj flkasjd flkasjf lkasjdf lajsdf ioajdflia lifajwliefja lisdjf alksdjf alkf over 30 omg omg omg omgo mgom ",
        date: "2018-02-13 20:03:17.035Z", 
        likes: 420,
        comments: [{
            userId: 2,
            avatar: "https://uproxx.files.wordpress.com/2017/02/kevin-durant-mad-yell.jpg?quality=95",
            userName: "KD",
            date: "4/4/2018",
            body: "your welcome",
        }, {
            userId: 2,
            avatar: "https://uproxx.files.wordpress.com/2017/02/kevin-durant-mad-yell.jpg?quality=95",
            userName: "KD",
            date: "2018-02-13 20:03:17.035Z",
            body: "u is welcome",
        }]
    }],
    comments: [{
        userId: 2,
        avatar: "https://uproxx.files.wordpress.com/2017/02/kevin-durant-mad-yell.jpg?quality=95",
        username: "KD",
        date: "4/4/2018",
        body: "i love this",
    }, {
        userId: 2,
        avatar: "https://uproxx.files.wordpress.com/2017/02/kevin-durant-mad-yell.jpg?quality=95",
        username: "KD",
        date: "2018-02-13 20:03:17.035Z",
        body: "omg",
    }]
}

describe('default page view', () => {
    it('should render App', () => {
        const component = shallow(<App />);
        expect(component).toHaveLength(1);
    });

    it('should render updates view by default', () => {
        const component = shallow(<App />);
        expect((component).find(Updates).length).toBe(1);
    });

    
});

describe('updates list view', () => {
    it('should limit words to 32 on update list items', () => {
        const component = mount(<Updates />);
        expect((component).find(UpdateListItem)).toHaveLength(1);
    });
});

describe('project comments feed', () => {
    it('should render 2 comments', () => {
        const component = mount(<CommentsFeed comments={testData.comments} />);
        expect((component).find(CommentItem)).toHaveLength(1);
    });
});