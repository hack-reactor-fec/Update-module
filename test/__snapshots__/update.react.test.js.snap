// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changeView method on App.jsx should change to comments view 1`] = `undefined`;

exports[`changeView method on App.jsx should render updatePostView if anything else is passed in other than updates & comments 1`] = `undefined`;

exports[`changeView method on App.jsx should set the state to whatever is passed in 1`] = `undefined`;
